client/src/mapView.js

```diff
@@ -10,15 +10,57 @@ var MapView = React.createClass({
     router: React.PropTypes.func
   },
 
-  render () {
+  // this is called after the first reder of the component
+  componentDidMount () {
     var {origin, destination} = this.context.router.getCurrentParams();
 
+    var start = this.getLatLong(origin);
+    var end = this.getLatLong(destination);
+
+    var map = this.initializeMap(start);
+    this.calcRoute(start, end, map);
+  },
+
+  // turns a lat/long string into a google maps LatLong Object
+  getLatLong (location) {
+    return new google.maps.LatLng(location.split(',')[0], location.split(',')[1]);
+  },
+
+  // initializes a map and attaches it to the map div
+  initializeMap (center) {
+    var mapOptions = {
+      zoom: 8,
+      center,
+    };
+    return new google.maps.Map(document.getElementById('map'), mapOptions);
+  },
+
+  // this creates a directions route from the start point to the end point
+  calcRoute (start, end, map) {
+    var directionsDisplay = new google.maps.DirectionsRenderer();
+    directionsDisplay.setMap(map);
+    var directionsService = new google.maps.DirectionsService();
+
+    var request = {
+      origin:start,
+      destination:end,
+      travelMode: google.maps.TravelMode.DRIVING
+    };
+    directionsService.route(request, function(result, status) {
```

Comment:
I think it would be a POST request since you are sending data along that is needed for the result evaluation